swri_image_util/src/nodelets/crosshairs_nodelet.cpp

```diff
@@ -0,0 +1,85 @@
+//
+// Copyright (c) 2017, Southwest Research Institute® (SwRI®)
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Southwest Research Institute® (SwRI®) nor the
+//       names of its contributors may be used to endorse or promote products
+//       derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL <COPYRIGHT HOLDER> BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+// LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+// ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// *****************************************************************************
+
+#include <string>
+
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <ros/ros.h>
+#include <nodelet/nodelet.h>
+#include <image_transport/image_transport.h>
+#include <sensor_msgs/image_encodings.h>
+#include <sensor_msgs/Image.h>
+#include <cv_bridge/cv_bridge.h>
+
+namespace swri_image_util
+{
+  class CrosshairsNodelet : public nodelet::Nodelet
+  {
+  public:
+    CrosshairsNodelet()
+    {
+    }
+
+    ~CrosshairsNodelet()
+    {
+    }
+
+    void onInit()
+    {
+      ros::NodeHandle &node = getNodeHandle();
+      ros::NodeHandle &priv = getPrivateNodeHandle();
+
+      image_transport::ImageTransport it(node);
+      image_pub_ = it.advertise("crosshairs_image", 1);
+      image_sub_ = it.subscribe("image", 1, &CrosshairsNodelet::ImageCallback, this);
+    }
+
+    void ImageCallback(const sensor_msgs::ImageConstPtr& image)
+    {
+      cv_bridge::CvImagePtr cv_image = cv_bridge::toCvCopy(image);
+      const int h = cv_image->image.rows;
+      const int w = cv_image->image.cols;
+      const cv::Scalar black(0, 0, 0);
+      cv::line(cv_image->image, cv::Point(0, w/2), cv::Point(h-1, w/2), black, 3, 8);
```

Comment:
Do these 3 and 8 mean something? I'm not sufficiently familiar with OpenCV to know the `cv::line()` syntax. You might prefer

```
const int thickness = 3;
const int line_type = 8; // 8-connected line
```

for clarity.